include/seqan3/utility/simd/detail/simd_algorithm_avx512.hpp

```diff
@@ -253,17 +253,33 @@ constexpr target_simd_t upcast_unsigned_avx512(source_simd_t const & src)
     }
 }
 
-// TODO: not implemented and used yet, if you implement it don't forget to add it to seqan3::detail::extract_half
 template <uint8_t index, simd::simd_concept simd_t>
-constexpr simd_t extract_half_avx512(simd_t const & src);
+constexpr simd_t extract_half_avx512(simd_t const & src)
+{
+    return reinterpret_cast<simd_t>(_mm512_castsi256_si512(
+            _mm512_extracti64x4_epi64(reinterpret_cast<__m512i const &>(src), index)));
+}
 
-// TODO: not implemented and used yet, if you implement it don't forget to add it to seqan3::detail::extract_quarter
+#if defined(__AVX512DQ__)
 template <uint8_t index, simd::simd_concept simd_t>
-constexpr simd_t extract_quarter_avx512(simd_t const & src);
+constexpr simd_t extract_quarter_avx512(simd_t const & src)
```

Comment:
Why are `extract_quarter/eighth_avx512` wrapped in `#if defined(__AVX512DQ__)` but `extract_half_avx512` isn't?

Reply:
The tests existed already (for SSE and AVX2). I ran the tests on icebear with AVX512 support to check the results.

The additional check is because the extract quarter requires an intrinsic that is only available by the AVX-512 CD subset. Not all AVX512 platforms might have this additional intrinsics subset, but still you can use other AVX512 intrinsics. The extract half doesn't need it because it only works with AVX512-F which is the foundational intrinsics set used by all that support AVX512.